Client/Frontend/Browser/Browser.swift

```diff
@@ -142,22 +142,29 @@ class Browser: NSObject, BrowserWebViewDelegate {
         // we extract the information needed to restore the tabs and create a NSURLRequest with the custom session restore URL
         // to trigger the session restore via custom handlers
         if let sessionData = self.sessionData {
-            restoring = true
+            if AppConstants.MOZ_PERSIST_SESSION_HISTORY {
+                restoring = true
```

Comment:
A potential concern is that the `restoring` property is referenced in the BVC for signalling when restoration is complete [1] and to check load state [2]. 

[1] https://github.com/mozilla/firefox-ios/blob/master/Client/Frontend/Browser/BrowserViewController.swift#L2515
[2] https://github.com/mozilla/firefox-ios/blob/master/Client/Frontend/Browser/BrowserViewController.swift#L840